packages/hooks/src/base.ts

```diff
@@ -49,28 +71,39 @@ export type MiddlewareCollector<T = any> = (self: any, fn: any, args: any[]) =>
  */
 export interface FunctionHookOptions<T = any> {
   middleware: Array<Middleware<T>>;
-  context: ContextUpdater<T>;
+  context: Array<ContextUpdater<T>>;
```

Comment:
Can we make this `Array<ContextUpdater<T>> | ContextUpdater<T>`? That way it stays backwards compatible for now (we can totally look at breaking/final API changes in a bit).

Reply:
FunctionHookOptions is the result of normalizeOptions, it is always an array.

The input is HookSettings and allows both context and context[], it is backward compatible and tested.